### PR TITLE
Remove some unnecessary HTML checks #5156

### LINKED ISSUE #5156: Remove some unnecessary HTML checks
As per title. They are present in a few UI tests.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #5156 

Explanations, going from first commit:

1. The same file is tested in `AllAccessControlUiTests`
2. As per TODO, but also this is a preparation for 3
3. All are tested in `AllAccessControlUiTests`
4. Full HTML check for the said page is already done before

### CODE REVIEW

File: src/test/java/teammates/test/cases/ui/browsertests/StudentProfilePageUiTest.java [hunk 1]
```diff
@@ -15,8 +15,13 @@
 import teammates.logic.backdoor.BackDoorServlet;
 import teammates.test.driver.BackDoor;
 import teammates.test.driver.TestProperties;
+import teammates.test.pageobjects.AppPage;
 import teammates.test.pageobjects.Browser;
 import teammates.test.pageobjects.BrowserPool;
+import teammates.test.pageobjects.EntityNotFoundPage;
+import teammates.test.pageobjects.GenericAppPage;
```
Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/StudentProfilePageUiTest.java]: Is this being used in this page?

File: src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java [hunk 2]
```diff
@@ -157,7 +158,7 @@ public void testPubliclyAccessiblePages() throws Exception {
         ______TS("unauthorized page");
         AppUrl url = createUrl(Const.ViewURIs.UNAUTHORIZED);
         currentPage.navigateTo(url);
-        verifyRedirectToNotAuthorized();
+        currentPage.verifyHtml("/unauthorized.html");
```
Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java]: can we combine these two lines to `verifyRedirectToNotAuthorized(url);`?

Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java]: I'd say it's redundant to do > 1 HTML check for the same page. Also, this is to make `testPubliclyAccessiblePages` look uniform.

Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/AllAccessControlUiTests.java]: I see, ok.

**Pull request merged**